sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-factory.ts

```diff
@@ -1,48 +1,61 @@
-import { MSFAANumber } from "@sims/sims-db";
+import { MSFAANumber, OfferingIntensity, Student } from "@sims/sims-db";
 import {
   E2EDataSources,
+  MSFAAStates,
   createFakeMSFAANumber,
   createFakeStudent,
   createFakeUser,
   saveFakeApplication,
   saveFakeStudent,
 } from "@sims/test-utils";
-import { MSFAATestInputData } from "./msfaa-part-time-process-integration.scheduler.models";
+import { MSFAATestInputData } from "./msfaa-helper";
 import { createFakeSINValidation } from "@sims/test-utils/factories/sin-validation";
 
 /**
  * Save an MSFAA record providing all data that will be
  * part of the MSFAA file generation.
  * @param db data source helper.
  * @param msfaa test input data.
+ * @param relations
+ * - `fakeStudent` fakeStudent to be associated with the MSFAATestInputData.
+ * @param options to saveMsfaaTestInputData.
+ * - `state` if provided, specifies the MSFAA State to be associated with the provided MSFAATestInputData.
+ * - `offeringIntensity` if provided, specifies the offering intensity associated with the provided MSFAATestInputData.
  * @returns a saved MSFAA record that uses the input test
  * data to be created.
  */
 export async function saveMSFAATestInputData(
   db: E2EDataSources,
   msfaaDataInput: MSFAATestInputData,
+  relations?: { fakeStudent?: Student },
+  options?: { state?: MSFAAStates; offeringIntensity?: OfferingIntensity },
 ): Promise<MSFAANumber> {
   // User.
   const fakeUser = createFakeUser();
   fakeUser.firstName = msfaaDataInput.givenName;
   fakeUser.lastName = msfaaDataInput.surname;
   fakeUser.email = msfaaDataInput.email;
   // Student.
-  const fakeStudent = createFakeStudent(fakeUser);
-  fakeStudent.birthDate = msfaaDataInput.birthDate;
-  fakeStudent.gender = msfaaDataInput.gender;
-  fakeStudent.contactInfo = {
-    address: {
-      addressLine1: msfaaDataInput.addressLine1,
-      addressLine2: msfaaDataInput.addressLine2,
-      city: msfaaDataInput.city,
-      country: msfaaDataInput.country,
-      selectedCountry: msfaaDataInput.country,
-      provinceState: msfaaDataInput.provinceState,
-      postalCode: msfaaDataInput.postalCode,
-    },
-    phone: msfaaDataInput.phone,
-  };
+  let fakeStudent: Student;
```

Comment:
maybe you could do something like this,
![image](https://github.com/bcgov/SIMS/assets/77353155/101157ae-b10d-4bb9-ae56-aad93578fd1b)

Reply:
Please change the assignment inside the condition, as @ann-aot mentioned, it would be better to refactor.

Reply:
I think @ann-aot meant exactly what she printed.
![image](https://github.com/bcgov/SIMS/assets/78114138/b5156d14-f304-41d8-a92b-7ab623c843cc)

Reply:
yes @andrepestana-aot. So, that we can avoid the `if else` conditions and `fakeStudent` can be a const too

Reply:
I am not using this function anymore and hence reverted this part of the code to how it was before.